controllers/operands/wasp.go

```diff
@@ -0,0 +1,449 @@
+	return dsConditionsToK8s(cr.(*appsv1.DaemonSet).Status.Conditions)
+}
+
+func NewWasp(hc *hcov1beta1.HyperConverged) *appsv1.DaemonSet {
+	waspImage, _ := os.LookupEnv(hcoutil.WaspImageEnvV)
+	wasp := NewWaspWithNameOnly(hc)
+	wasp.Namespace = hc.Namespace
+	wasp.Spec = appsv1.DaemonSetSpec{
+		Selector: &metav1.LabelSelector{
+			MatchLabels: map[string]string{
+				"name": "wasp",
+			},
+		},
+		UpdateStrategy: appsv1.DaemonSetUpdateStrategy{
+			Type: appsv1.RollingUpdateDaemonSetStrategyType,
+			RollingUpdate: &appsv1.RollingUpdateDaemonSet{
+				MaxUnavailable: ptr.To(intstr.FromString("10%")),
+				MaxSurge:       ptr.To(intstr.FromString("0%")),
+			},
+		},
+		Template: corev1.PodTemplateSpec{
+			ObjectMeta: metav1.ObjectMeta{
+				Labels: map[string]string{
+					"name": "wasp",
+				},
+			},
+			Spec: corev1.PodSpec{
+				RestartPolicy:                 corev1.RestartPolicyAlways,
+				ServiceAccountName:            waspServiceAccount,
+				HostPID:                       true,
+				HostUsers:                     ptr.To(true),
+				TerminationGracePeriodSeconds: ptr.To(int64(30)),
+				Containers: []corev1.Container{
+					{
+						Name:            "wasp-agent",
+						Image:           waspImage,
+						ImagePullPolicy: corev1.PullIfNotPresent,
+						Env: []corev1.EnvVar{
+							{
+								Name:  "FSROOT",
+								Value: "/host",
+							},
+							{
+								Name:  "SWAPINNES",
+								Value: "5",
+							},
+							{
+								Name:  "LOOP",
+								Value: "true",
+							},
+							{
+								Name:  "DEBUG",
+								Value: "",
+							},
+						},
+						Resources: corev1.ResourceRequirements{
+							Requests: corev1.ResourceList{
+								corev1.ResourceCPU:    resource.MustParse("100m"),
+								corev1.ResourceMemory: resource.MustParse("50M"),
+							},
+						},
+						SecurityContext: &corev1.SecurityContext{
+							Privileged: ptr.To(true),
+						},
+						VolumeMounts: []corev1.VolumeMount{
+							{
+								Name:      "host",
+								MountPath: "/host",
+							},
+						},
+					},
+				},
+				Volumes: []corev1.Volume{
+					{
+						Name: "host",
+						VolumeSource: corev1.VolumeSource{
+							HostPath: &corev1.HostPathVolumeSource{
+								Path: "/",
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	injectPlacement(hc.Spec.Workloads.NodePlacement, &wasp.Spec)
+	if _, ok := hc.Annotations[waspDryRunAnnotation]; ok {
+		wasp.Spec.Template.Spec.Containers[0].Env =
+			append(wasp.Spec.Template.Spec.Containers[0].Env,
+				corev1.EnvVar{
+					Name:  "DRY_RUN",
+					Value: "true",
+				})
+	}
+	return wasp
+}
+
+func NewWaspWithNameOnly(hc *hcov1beta1.HyperConverged) *appsv1.DaemonSet {
+	return &appsv1.DaemonSet{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "wasp-agent",
+			Labels:    getLabels(hc, hcoutil.AppComponentWasp),
+			Namespace: hc.Namespace,
+		},
+	}
+}
+
+func dsConditionsToK8s(conditions []appsv1.DaemonSetCondition) []metav1.Condition {
```

Comment:
See comment above, on the `getConditions` method https://github.com/kubevirt/hyperconverged-cluster-operator/pull/2814/files#r1536748265.